pkg/analyzer/detection/detection.go

```diff
@@ -15,15 +16,27 @@ import (
 // LogLayerName is the name of the log field holding the detection target.
 const LogLayerName = "layer"
 
+// DetectComponentOpts contains namespace information for DetectComponents
```

Comment:
at the moment yes, but `opts` implies these are just options we can set to configure how we want the detection to work